frontend/lib/resolution/copy-elision.cpp

```diff
@@ -600,6 +692,9 @@ void FindElidedCopies::handleScope(const AstNode* ast, RV& rv) {
           CopyElisionState& parentState = parent->copyElisionState[id];
           parentState.lastIsCopy = true;
           parentState.points.insert(state.points.begin(), state.points.end());
+        } else if (frame->eligibleVars.count(id) == 0) {
+          CopyElisionState& parentState = parent->copyElisionState[id];
+          parentState.lastIsCopy = false;
```

Comment:
Note that prior, the invariant has been that if the last reference is not a copy, the list of `points` is emptied. You probably want to maintain this invariant, lest this list get polluted with older elision points.